browser/scripts/player.js

```diff
@@ -34,9 +34,57 @@ function Player() {
 	this.core.active_graph = this.core.root_graph = new Graph(this.core, null, 'root')
 	this.core.graphs.push(this.core.root_graph)
 
+	var canvas = E2.dom.webgl_canvas[0]
+	this.vrControlCamera = new THREE.PerspectiveCamera(
+				90,
+				canvas.clientWidth / canvas.clientHeight,
+				0.001,
+				1000)
+
+	this.concatenatedCamera = this.vrControlCamera.clone()
+
+	this.setupInput()
+
 	this.boundOnAnimFrame = this.on_anim_frame.bind(this)
 }
 
+Player.prototype.setupInput = function() {
+	this.rayInput = new RayInput.default(this.concatenatedCamera, E2.dom.webgl_canvas[0])
+	
+	var el = document.body
+
+	this.rayInput.on('raydown', function(mesh) {
+		if (!mesh || E2.app.worldEditor.isActive())
+			return;
+
+		E2.core.runtimeEvents.emit('gazeClicked:'+mesh.uuid)
```

Comment:
@gmarinov this could be renamed gazeDown, perhaps, but it would mean making a new plugin for gazeUp, or renaming gaze to something ... this keeps the old functionality as is, just with Ray Input.